pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -748,6 +842,200 @@ func (oe *operationExecutor) VerifyControllerAttachedVolume(
 		volumeToMount.VolumeName, "" /* podName */, verifyControllerAttachedVolumeFunc, opCompleteFunc)
 }
 
+// VolumeStateHandler defines a set of operations for handling mount/unmount/detach/reconstruct volume-related operations
+type VolumeStateHandler interface {
```

Comment:
How is this different then `OperationExecutor` itself and why is it needed?

Reply:
This is wrapper to choose corresponding volume operations based on volumeMode.
To avoid adding `if volumeMode = "Block" {abc} else {xyz}` everywhere, we introduced this handler to select volumehandler from volumeSpec.